pkg/resourceskeleton/testdata/uri-skeleton.yaml

```diff
@@ -1111,34 +1111,30 @@
     location: us-central1
   ResourceConfigId: EdgeContainerVpnConnection
   URI: "https://edgecontainer.googleapis.com/v1/projects/kcc-test/locations/us-central1/vpnConnections/edgecontainervpnconnection-test"
-- ExpectedSkeleton:
-    apiVersion: vertexai.cnrm.cloud.google.com/v1beta1
-    kind: VertexAIDataset
-    spec:
-      projectRef:
-    external: kcc-test
-    location: us-central1
+- ExpectedSkeleton: null # VertexAIDataset does not support import
   ResourceConfigId: VertexAIDataset
-  URI: "https://us-central1-aiplatform.googleapis.com/v1beta1/projects/kcc-test/locations/us-central1/datasets/1234567890?alt=json"
+  URI: ""
```

Comment:
Looking at other examples, URI may or may not be kept when ExpectedSkeleton is null. Do you mind adding a unit test to ensure the behavior is consistent? I think either not setting it (if possible) or setting it to an empty string should work.

Reply:
Related to the previous comment, the test `TestNewFromURI` is currently skipped for `VertexAIDataset`. Do we still need URI?

Reply:
I don't think we need it!

Reply:
Just double check on my understanding. I saw that resource `VertexAIDataset` does not support import. Therefore we should set `idTemplateCanBeUsedToMatchResourceName` to `false` for this resource?

https://registry.terraform.io/providers/hashicorp/google-beta/4.84.0/docs/resources/vertex_ai_dataset#import

Reply:
In most cases, you are correct. More specifically, the decision flow I have for this field is:

1. Do I need to support export for this resource? Yes: Go to step 2. No: Set the value to `false`.
2. Can the `IDTemplate` be used to match the resource's URL? Yes: Set the value to `true`. No: Go to step 3.
3. Figure out the export use case / design for this resource.

Whether TF support import of this resource or not can be helpful to answer the question in step 2 above.